Assets/Scripts/Player/WeaponInventory.cs

```diff
@@ -118,13 +132,13 @@ private void Update()
         }
 
         // Drop weapon.
-        if (Input.GetKeyDown(KeyCode.Q))
+        if (Input.GetKeyDown(KeyCode.Q) && !playerGun.reloading)
         {
             RemoveWeapon(currentWeaponIndex);
         }
     }
 
-    private void ChangeWeapon(int newWeaponIndex, bool force = false)
+    private void ChangeWeapon(int newWeaponIndex, bool force = false, bool removed = false)

```

Comment:
What does `removed` signal here? It looks like it's only called once when `force` is also `true`. Is there any difference here?